dotcom-rendering/src/components/ShareButton.importable.tsx

```diff
@@ -0,0 +1,233 @@
+				fill: ${themePalette('--share-button-liveblog-mobile')};
+			}
+		}`}
+	}
+`;
+
+const getUrl = ({
+	pageId,
+	CMP,
+	blockId,
+}: {
+	pageId: string;
+	CMP?: string;
+	blockId?: string;
+}) => {
+	const searchParams = new URLSearchParams({});
+	if (CMP) searchParams.append('CMP', CMP); // Do we want to track this?
+	if (blockId) searchParams.append('page', `with:block-${blockId}`);
+
+	const blockHash = blockId ? `#block-${blockId}` : '';
+	const paramsString = searchParams.toString();
+	return new URL(
+		`${pageId}${paramsString ? '?' + paramsString : ''}${blockHash}`,
+		'https://www.theguardian.com/',
+	).href;
+};
+
+export const NativeShareButton = ({
+	onShare,
+	size,
+	isLiveBlogMeta,
+}: {
+	onShare: () => void;
+	size?: Size;
+	isLiveBlogMeta: boolean;
+}) => {
+	const sizeXSmall = size === 'xsmall';
+
+	return (
+		<Button
+			onClick={() => onShare()}
+			size={size}
+			type="button"
+			priority="tertiary"
+			iconSide="left"
+			icon={<SvgShare />}
+			css={[
+				buttonStyles,
+				nativeShare(sizeXSmall),
+				isLiveBlogMeta && liveBlogMobile(false),
+			]}
+		>
+			Share
+		</Button>
+	);
+};
+
+export const CopyLinkButton = ({
+	onShare,
+	size,
+	isLiveBlogMeta,
+	isCopied,
+}: {
+	onShare: () => void;
+	size?: Size;
+	isLiveBlogMeta: boolean;
+	isCopied: boolean;
+}) => {
+	const sizeXSmall = size === 'xsmall';
+	return (
+		<Button
+			onClick={() => onShare()}
+			size={size}
+			type="button"
+			priority="tertiary"
+			iconSide="left"
+			icon={isCopied ? <SvgCheckmark /> : <LinkIcon />}
+			css={[
+				...(isCopied
+					? [copiedButtonStyles(sizeXSmall)]
+					: [buttonStyles]),
+				sharedButtonStyles(sizeXSmall),
+				isLiveBlogMeta && liveBlogMobile(isCopied),
+			]}
+		>
+			{isCopied ? 'Link copied' : 'Copy link'}
+		</Button>
+	);
+};
+
+export const ShareButton = ({
+	size = 'small',
+	pageId,
+	blockId,
+	webTitle,
+	format,
+	context,
+}: Props) => {
+	const [isCopied, setIsCopied] = useState(false);
+	const [isShareSupported, setIsShareSupported] = useState(false);
+
+	const isLiveBlogMeta =
+		(format.design === ArticleDesign.LiveBlog &&
+			context === 'ArticleMeta') ||
+		context === 'SubMeta';
+
+	const shareData = {
+		title: `${webTitle}`,
+		text: `${webTitle}:`,
+		url: getUrl({
+			pageId,
+			blockId,
+		}),
+	};
+
+	useEffect(() => {
+		setIsShareSupported(
+			typeof navigator !== 'undefined' &&
+				'share' in navigator &&
+				navigator.canShare(shareData),
+		);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, []);
+
+	useEffect(() => {
+		if (!isCopied) return;
+		const timer = setTimeout(() => {
+			setIsCopied(false);
+		}, 3000);
+		return () => clearTimeout(timer);
+	}, [isCopied]);
+
+	return isShareSupported ? (
+		<NativeShareButton
+			onShare={() => {
+				navigator.share(shareData).catch(() => {});
+			}}
+			size={size}
+			isLiveBlogMeta={isLiveBlogMeta}
+		/>
+	) : (
+		<CopyLinkButton
+			onShare={() => {
+				if (!('clipboard' in navigator)) return;
```

Comment:
If the clipboard isn't available should we hide the button entirely? Otherwise I think we have a button that says "Copy link" but doesn't work?

So essentially we have three levels depending on features available:

1. If share is supported show a share button
2. Else if clipboard is supported show a copy button
3. Else show nothing